homeassistant/components/mqtt/util.py

```diff
@@ -136,12 +136,9 @@ def valid_birth_will(config: ConfigType) -> ConfigType:
     return config
 
 
-def get_mqtt_data(hass: HomeAssistant, ensure_exists: bool = False) -> MqttData:
+def get_mqtt_data(hass: HomeAssistant) -> MqttData:
     """Return typed MqttData from hass.data[DATA_MQTT]."""
     mqtt_data: MqttData
-    if ensure_exists:
```

Comment:
Instead of just removing the check, maybe raise instead:
```py
if DATA_MQTT not in hass.data:
    raise HomeAssistantError
```

Or maybe the `KeyError` is good enough?

Reply:
Key Error should be enough I think